client/client.go

```diff
@@ -78,6 +78,12 @@ func New(cfg *config.Config) (*DatabricksClient, error) {
 	}, nil
 }
 
+type DatabricksClientInterface interface {
+	Do(ctx context.Context, method, path string,
+		headers map[string]string, request, response any,
+		visitors ...func(*http.Request) error) error
+}
+
```

Comment:
this is not the right interface to expose - if we do a change, then it has to be 

```
type ApiClientDoer interface {
   Do(ctx context.Context, method, path string, opts ...httpclient.DoOption) error
}
```

https://github.com/databricks/databricks-sdk-go/blob/main/httpclient/api_client.go#L100-L101